feat(sed): support multiple file names

nfischer opened the pull request:
fixes #231. Semantics are like unix sed.

Also, if we merge this, it makes the PR to add support for pipes slightly nicer.

ariporad commented on src/sed.js:
```diff
@@ -32,19 +33,32 @@ function _sed(options, regex, replacement, file) {
-  if (!file)
-    common.error('no file given');
+  if (!files)
+    common.error('no paths given');
```
Small nitpick: maybe make this "no files given"?

nfischer replied:
Done

ariporad commented on src/sed.js:
```diff
@@ -32,19 +33,32 @@ function _sed(options, regex, replacement, file) {
-    fs.writeFileSync(file, result, 'utf8');
+  var sed = [];
+  files.forEach(function(file) {
+    if (!fs.existsSync(file)) {
```
I don't think we need this, I think glob filters out nonexistent files. (Sorry, on mobile, so I can't check.)

nfischer replied:
The current `expand()` will not filter out nonexistent files, so I think this should still be in place. For example:

```Javascript
> sed('search', 'replacement', 'nonexistentfilename');
sed: no such file or directory: nonexistentfilename
```

ariporad commented on test/sed.js:
```diff
@@ -74,4 +74,24 @@ var result = shell.sed('l*\\.js', '', 'resources/grep/file');
+assert.equal(shell.error(), null);
+assert.equal(result, 'hello1\nhello2');
+assert.equal(shell.cat('tmp/file1'), 'hello1');
+assert.equal(shell.cat('tmp/file2'), 'hello2');
```
Can we add some tests for regexes?

nfischer replied:
I modified one of these tests to use the regex `/t.*st/` instead of `'test'`. It shouldn't change anything, since this doesn't modify how regexes are handled.

ariporad commented:
I have a few notes, but it otherwise looks good.

nfischer commented:
@ariporad I addressed your comments. I also added an invalid test case, since I realized I forgot to add one of those  :+1:

And I removed the leading slash from one of the existing test cases to play better with Windows compatibility (not sure if it was an issue, but it should at least make the test more robust, since it doesn't depend on the user's root directory contents anymore).

ariporad commented:
LGTM!

nfischer commented:
LGTM. Merging

cvrebert commented:
Superb! Is there any rough time horizon for the next release version of shelljs?

nfischer commented:
We've started tracking issues under the v0.6 milestone (that'll be our next release). Feel free to help out if you're available (or report any major bugs you think should be in the next release)

https://github.com/shelljs/shelljs/milestones/v0.6.0

cvrebert commented:
Thanks
